### PR TITLE
added n6 keyboard

### PR DESCRIPTION
**Pull request by yulei**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
is there a reason this uses rgblight instead of rgb matrix?

**Comment by yulei** [comment]
> is there a reason this uses rgblight instead of rgb matrix?

It's a seperate rgb bar, not the per key rgb led.

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/neson_design/n6/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes		# Enable keyboard RGB underglow
+
+SRC += i2c_master.c issi/is31fl3731.c
```
Comment (fauxpark) [inline keyboards/neson_design/n6/rules.mk:22]:
```suggestion
QUANTUM_LIB_SRC += i2c_master.c issi/is31fl3731.c
```

File: keyboards/neson_design/n6/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+A 65% keyboard from the Neson Design's N series.
+
+* Keyboard Maintainer: [astro](https://github.com/yulei)  
+* Hardware Supported: N6 keyboard
+* Hardware Availability: Limited gb
+
+Make example for this keyboard (after setting up your build environment):
+
+    make inett_studio/n6:default
```
Comment (fauxpark) [inline keyboards/neson_design/n6/readme.md:11]:
```suggestion
    make neson_design/n6:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**